### PR TITLE
Fix for as_handle exception propagation

### PR DESCRIPTION
**Pull request by bow**
This pull request addresses issue #1544 

I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

I am happy be thanked by name in the ``NEWS.rst`` and ``CONTRIB.rst`` files,
and have added myself to those files as part of this pull request. (*This
acknowledgement is optional. Note we list the names sorted alphabetically.*)

### CODE REVIEW

File: Bio/File.py [hunk 1]
```diff
@@ -82,16 +82,23 @@ def as_handle(handleish, mode='r', **kwargs):
     been deprecated (this happens automatically in text mode).
 
     """
-    try:
-        if sys.version_info[0] >= 3 and "U" in mode:
-            mode = mode.replace("U", "")
+    # Workaround for https://github.com/biopython/biopython/issues/1544.
+    # TODO: Remove when Python 3.5 dependency is dropped.
+    if sys.version_info[:2] < (3, 6):
```
Comment (peterjc) [inline Bio/File.py]: I think this needs to be also checking at least Python 3? Otherwise:

```python
ImportError: No module named pathlib
```

Comment (peterjc) [reply Bio/File.py]: Correction: At least Python 3.4
https://docs.python.org/3.4/library/pathlib.html

Specifically we don't want to do this on Python 2

Comment (bow) [reply Bio/File.py]: Ah, indeed ~ I missed the version 2 check. It's updated now, also with a more granular check for the minor version.

**Comment by bow** [comment]
Closed in favor of #1555

**Pull request closed without merging**